### PR TITLE
Yzj ui update

### PR DESCRIPTION
**Pull request by yzj1212**
UI & admin tool finished.

### CODE REVIEW

File: BaseModel/src/com/wpi/cs509/teamA/ui/AdminDialog.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+import javax.swing.JTextField;
+
+/**
+ * This is the class that administrators uses to log in
+ * @author ZYang
+ *
+ */
+public class AdminDialog extends JDialog implements ActionListener {
+
+	private final JPanel contentPanel = new JPanel();
```
Comment (sbvictory) [inline BaseModel/src/com/wpi/cs509/teamA/ui/AdminDialog.java]: Why final here?

File: BaseModel/src/com/wpi/cs509/teamA/ui/AdminDialog.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	private JPasswordField passwordField;
+	private JButton okButton;
+	private JButton cancelButton;
+	private ImageComponent imgPanel;
+	
+	//private JFrame controllingFrame;
+	/**
+	 * Launch the application.
+	 */
+	public static void main(String[] args) {
```
Comment (sbvictory) [inline BaseModel/src/com/wpi/cs509/teamA/ui/AdminDialog.java]: Remove

File: BaseModel/src/com/wpi/cs509/teamA/ui/AdminDialog.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	public AdminDialog(ImageComponent imageComponent) {
+		this.imgPanel = imageComponent;
+		setTitle("Log in");
+		setBounds(100, 100, 450, 300);
+		getContentPane().setLayout(new BorderLayout());
+		contentPanel.setBorder(new EmptyBorder(5, 5, 5, 5));
+		getContentPane().add(contentPanel, BorderLayout.CENTER);
+		contentPanel.setLayout(null);
+		
+		JLabel lblPassword = new JLabel("Password:");
```
Comment (sbvictory) [inline BaseModel/src/com/wpi/cs509/teamA/ui/AdminDialog.java]: Add some comments to every part to clarify what every part for.

File: BaseModel/src/com/wpi/cs509/teamA/ui/AdminDialog.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+		passwordField = new JPasswordField(10);
+		passwordField.setBounds(174, 76, 126, 28);
+		passwordField.setColumns(10);
+		contentPanel.add(passwordField);
+		
+		JPanel buttonPane = new JPanel();
+		buttonPane.setLayout(new FlowLayout(FlowLayout.RIGHT));
+		getContentPane().add(buttonPane, BorderLayout.SOUTH);
+		
+		okButton = new JButton("OK");
```
Comment (sbvictory) [inline BaseModel/src/com/wpi/cs509/teamA/ui/AdminDialog.java]: String final OK = "OK";

File: BaseModel/src/com/wpi/cs509/teamA/ui/AdminDialog.java [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+		okButton.setActionCommand("OK");
+		okButton.addActionListener(this);
+		buttonPane.add(okButton);
+		getRootPane().setDefaultButton(okButton);
+		
+		cancelButton = new JButton("Cancel");
```
Comment (sbvictory) [inline BaseModel/src/com/wpi/cs509/teamA/ui/AdminDialog.java]: private String final CANCEL= "Cancel";

File: BaseModel/src/com/wpi/cs509/teamA/ui/AdminMouseListener.java [hunk 6]
```diff
@@ -37,28 +44,65 @@ public AdminMouseListener() {
...
+			//	JOptionPane.showMessageDialog(null,
+		    //            "Please select text field");
+			//}else{
+				neighborDialog.setFieldTitle(e.getX(), e.getY());
+			//}
+		}else if(e.getButton()==MouseEvent.BUTTON1&&!neighborDialog.isVisible()){
+			xPos = e.getX();
+			yPos = e.getY();
+	
+			System.out.println("This click from admin user..");
```
Comment (sbvictory) [inline BaseModel/src/com/wpi/cs509/teamA/ui/AdminMouseListener.java]: Comments..

File: BaseModel/src/com/wpi/cs509/teamA/ui/NodeManageDialog.java [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+/**
+ * This is the class that administrators uses to add nodes to database
+ * @author ZYang
+ *
+ */
+
+@SuppressWarnings("serial")
+public class NodeManageDialog extends JDialog implements ActionListener {
+
+	private final JPanel contentPanel = new JPanel();
```
Comment (sbvictory) [inline BaseModel/src/com/wpi/cs509/teamA/ui/NodeManageDialog.java]: No final

File: BaseModel/src/com/wpi/cs509/teamA/ui/UserScreen.java [hunk 8]
```diff
@@ -38,35 +44,69 @@ private UserScreen() {
...
+		neighborDialog = new NeighborDialog();
+		neighborDialog.setVisible(false);
 
 		// the panel to show image
-		imgComponent = new ImageComponent(inputPanel);
+		imgComponent = new ImageComponent(inputPanel, this, neighborDialog);
+
 		// display the image
-		imgComponent.setImagePath(System.getProperty("user.dir") + "\\src\\CSP.jpg");
+		imgComponent.setImagePath(System.getProperty("user.dir") + "/src/Final_Campus_Map.jpg");
```
Comment (sbvictory) [inline BaseModel/src/com/wpi/cs509/teamA/ui/UserScreen.java]: make a comment here..

File: BaseModel/src/com/wpi/cs509/teamA/ui/AdminDialog.java (hunk 3 above)
Comment (yzj1212) [reply BaseModel/src/com/wpi/cs509/teamA/ui/AdminDialog.java:66]: OK, Brother one

**Pull request merged**